Implementing Shor's algorithm using the Qrisp framework on Cat Qubits.ipynb

```diff
@@ -0,0 +1,770 @@
+    "            p = math.gcd(x - 1, N)\n",
+    "            q = math.gcd(x + 1, N)\n",
+    "            if p != 1 and p != N:\n",
+    "                return (p, N // p)\n",
+    "            if q != 1 and q != N:\n",
+    "                return (q, N // q)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "e4a50a51",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Test passed!\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Test the classical Shor's algorithm for N=10000\n",
+    "N=10000\n",
+    "check_list = list()\n",
+    "for i in range(2,N):\n",
+    "    (p,q) = classical_Shor(i)\n",
+    "    check = (p*q == i)\n",
+    "    if(check==False):\n",
+    "        print(i, (p,q))\n",
+    "else:\n",
+    "    print('Test passed!')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "id": "01f21a4b",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "(3, 5)"
+      ]
+     },
+     "execution_count": 7,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "classical_Shor(15)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a0d13a57",
+   "metadata": {},
+   "source": [
+    "The factorisation of 15 only requires small fractions of a second to be achieved with the brute-force algorithm.\n",
+    "\n",
+    "NOTE: you can use the `%%timeit` magic command to measure the time it takes to run a cell in a Jupyter notebook, although this will increase the actual time needed to execute the cell since multiple runs are performed. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ce4d5d33",
+   "metadata": {},
+   "source": [
+    "#### Quantum Period finding\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "id": "9662c206-a627-4a79-b3e8-0abe336f9140",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sympy import continued_fraction_convergents, continued_fraction_iterator, Rational\n",
+    "\n",
+    "def get_r_candidates(approx):\n",
+    "    rationals = continued_fraction_convergents(continued_fraction_iterator(Rational(approx)))\n",
+    "    return [rat.q for rat in rationals]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "id": "81e211ed",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def get_phase_candidates(a, N):\n",
+    "    qg = QuantumModulus(N)\n",
+    "    qg[:] = 1\n",
+    "    qpe_res = QuantumFloat(2*qg.size + 1, exponent = -(2*qg.size + 1))\n",
+    "    h(qpe_res)\n",
+    "    x=a\n",
+    "    for i in range(len(qpe_res)):\n",
+    "        with control(qpe_res[i]):\n",
+    "            qg *= x\n",
+    "            x = (x*x)%N\n",
+    "    QFT(qpe_res, inv = True)\n",
+    "    qiskit_qc = qpe_res.qs.compile().to_qiskit() # compile the circuit to qiskit for technical limitations\n",
```

Comment:
The comment looks a bit out of place, could you add it on top of the line instead?